### ISSUE TITLE
.on() event not complete

### ISSUE THREAD

**Issue by eek** [HubSpot/youmightnotneedjquery#179]
One of the cool features jQuery has, is binding events to elements that do no exist yet.

```
document.on('click', '.selector', function() {
    console.log('click on a selector that might appear sometime');
}

```
This would be a great addition for the ON section, that is lacking the selector version

![image](https://user-images.githubusercontent.com/2953026/28935378-0bcb2204-788d-11e7-9cfd-ca74182fdfc7.png)

**Comment by medslt** [comment]
+1

**Comment by robsonsobral** [comment]
You mean an Event Delegation?

**Comment by radiocity** [comment]
ez!
```javascript
document.addEventListener("click", function (e) {
    if (e.target.classList && e.target.classList.contains("test")) {
        // Add some magic here
    }
});
```
[Demo](https://jsfiddle.net/k3xba70p/)

**Comment by radiocity** [comment]
ez!
```javascript
document.addEventListener("click", function (e) {
    if (e.target.matches(".selector")) {
        // Add some magic here
    }
});
```
[Demo](https://jsfiddle.net/vmov4z3r/1/)

**Comment by robsonsobral** [comment]
@radiocity, I think you can improve your code using [`.matches`](https://developer.mozilla.org/en-US/docs/Web/API/Element/matches).

**Comment by radiocity** [comment]
@robsonsobral 
Awesome! Didnt know. Now I can improve a lot  💩code of mine

**Comment by jeremejazz** [comment]
Good examples from [David Walsh](https://davidwalsh.name/event-delegate)

```js
// Get the element, add a click listener...
document.getElementById("parent-list").addEventListener("click", function(e) {
	// e.target is the clicked element!
	// If it was a list item
	if(e.target && e.target.nodeName == "LI") {
		// List item found!  Output the ID!
		console.log("List item ", e.target.id.replace("post-", ""), " was clicked!");
	}
});
```

and using the Element.matches API 

```js
// Get the parent DIV, add click listener...
document.getElementById("myDiv").addEventListener("click",function(e) {
	// e.target was the clicked element
  if (e.target && e.target.matches("a.classA")) {
    console.log("Anchor element clicked!");
	}
});
```

**Comment by Abbondanzo** [comment]
Closed by #262, thanks for the suggestion!
